torch/_functorch/aot_autograd.py

```diff
@@ -1254,5 +1261,79 @@ def _aot_export_function(
     return fx_g, meta, in_spec, out_spec.spec
 
 
+@contextmanager
+def _detect_attribute_assignment(mod: torch.nn.Module):
+    # Do not allow assignment of tensor attributes during export unless
+    # the attribute is registered as a buffer.
+
+    STD_ATTRS = {
+        "_backward_hooks",
+        "_backward_pre_hooks",
+        "_buffers",
+        "_forward_hooks",
+        "_forward_hooks_always_called",
+        "_forward_hooks_with_kwargs",
+        "_forward_pre_hooks",
+        "_forward_pre_hooks_with_kwargs",
+        "_is_full_backward_hook",
+        "_load_state_dict_post_hooks",
+        "_load_state_dict_pre_hooks",
+        "_modules",
+        "_non_persistent_buffers_set",
+        "_parameters",
+        "_state_dict_hooks",
+        "_state_dict_pre_hooks",
+        "training",
+    }
+
+    def _get_attributes(mod):
+        # return any attributes of a module that are not standard attributes
+        return {k: v for k, v in mod.__dict__.items() if k not in STD_ATTRS}
+
+    def _set(obj, kp, v):
+        *kp, k = kp
+        obj = pytree.key_get(obj, kp)
+        if isinstance(k, pytree.GetAttrKey):
+            setattr(obj, k.name, v)
+        elif isinstance(k, pytree.MappingKey):
+            obj[k.key] = v
+        elif isinstance(k, pytree.SequenceKey):
+            obj[k.idx] = v
```

Comment:
Does this work for tuples?

Reply:
Nice, it breaks with tuples. :) I'll do the restore differently.